### ISSUE TITLE
Create a guide on bean validation

### ISSUE THREAD

**Issue by NottyCode** [OpenLiberty/guides-common#8]
Write an Open Liberty guide for Bean Validation 2.0.

**Comment by alexsm82** [comment]
A draft outline of this guide:

-Title
-What you will learn
-Getting Started
-What is Bean Validation
-What’s new in Bean Validation 2.0
-Applying Constraints
   -Construct a bean with several different built-in constraints applied to several fields (mix of existing and new 2.0 constraints)
       -Include validating values and keys of a map
       -Include a nested bean to be validated
       -Include a custom constraint
-Building the application
-Testing the Constraints
   -Demonstrate validating the bean with valid/invalid data
-Great work! You’re done!

**Comment by yeekangc** [comment]
@alexsm82, you think 15 to 20 mins will be adequate for a user to work through all these?

Any thoughts around the scenario or sample application that you will use to take the users through all these?

**Comment by alexsm82** [comment]
@yeekangc Yes, I think they will be able to work through all of these in 15 to 20 minutes.  Most of the mentioned scenarios will be on the same bean, it will just require going through them and adding the appropriate constraints.  

In terms of the scenario, the most basic we could have would essentially just be the java bean classes that store/validate the data (I was thinking space-related data would match the Open Liberty theme) and then test those with various data inputs. We could, however, come up with a slightly more real life and complex scenario if you think that would be better.

**Comment by yeekangc** [comment]
Great, @alexsm82.

A space-themed sample will work.  If we can have more real life scenario that isn't complex, that will be great.

You can find more information on how to create a guide here:  https://github.com/OpenLiberty/guides-common/wiki/How-to-create-an-Open-Liberty-guide

**Comment by yeekangc** [comment]
Should target EE 8.

**Comment by yeekangc** [comment]
This is for Bean Validation 2.0 in Java EE 8.  Should align the guide with when the feature is released.

**Comment by yeekangc** [comment]
Initial drop by @alexsm82 here: https://github.com/OpenLiberty/draft-guide-beanvalidation/pull/1

**Comment by evelinec** [comment]
@sishida can you start formal review on this guide pls? Thank you. 

Repo link: https://github.com/OpenLiberty/draft-guide-beanvalidation

**Comment by sishida** [comment]
Link to ID review: https://github.com/OpenLiberty/draft-guide-beanvalidation/pull/13
